packages/tupaia-web/src/features/Visuals/Matrix/Matrix.tsx

```diff
@@ -176,7 +180,8 @@ const MatrixVisual = () => {
   return (
     <>
       <MatrixComponent
-        {...config}
+        // casting here because we know that the config is a MatrixConfig and it has a different shape than configs of other types, and while we know that this component only ever gets a MatrixConfig, the Matrix component doesn't know that as it all comes from the same context
+        {...(config as MatrixConfig)}
```

Comment:
I didn't really want to have to cast the type here, but I wasn't sure of a better way, since all the config is from a common context that has a union type for config